### PR TITLE
treewide: fix links to `specifications.freedesktop.org`

### PR DESCRIPTION
**Pull request by zvictor**
Links to `specifications.freedesktop.org` docs are broken.
This PR fixes them all.

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ convertIconTheme() {
     local -ra scales=([1]="" [2]="@2")
 
     # Based loosely on the algorithm at:
-    # https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#icon_lookup
+    # https://specifications.freedesktop.org/icon-theme-spec/latest/icon_lookup
```
Comment (philiptaron) [inline pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh:40]: broken link

**Review by philiptaron — CHANGES_REQUESTED** [review]
All but one link is good 👍🏻

File: pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh (hunk 1 above)
Comment (zvictor) [reply pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh:40]:
```suggestion
    # https://specifications.freedesktop.org/icon-theme-spec/latest/#icon_lookup
```

**Comment by zvictor** [comment]
> All but one link is good 👍🏻

It should be fixed now. Thank you for verifying!

**Review by philiptaron — APPROVED** [review]

**Pull request merged**